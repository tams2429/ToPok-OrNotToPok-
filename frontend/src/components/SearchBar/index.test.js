import { render } from '@testing-library/react'
import { SearchBar } from './' 




test('Search Bar component', () => {

  const mockSetSearchString = jest.fn()
  const mockHandleSearch = jest.fn()
  const initialSearchString = ''

  const { searchBarComponent } = render(<SearchBar searchString={initialSearchString} setSearchString={mockSetSearchString} handleSearch={mockHandleSearch}/>)

  //Test that the component renders properly and matches snapshot
  it('should render properly and match snapshot', () => {
    expect(searchBarComponent).toMatchSnapshot()
  })
})

//Test to check that mockSetSearchString (i.e. mockFunction) is being invoked with the input value, as user is typing

//Test that pressing the enter key calls the mockHandleSearch (i.e. mockFunction) + with the current searchString (mockSearchString)?