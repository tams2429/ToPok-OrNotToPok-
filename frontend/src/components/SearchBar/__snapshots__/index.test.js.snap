// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Bar component should render properly and match snapshot 1`] = `
<div>
  <input
    aria-label="Searchbar-for-pokemon"
    class="searchBar"
    data-testid="searchBar"
    placeholder="Search your PokÃ©mon here..."
    type="search"
    value=""
  />
</div>
`;
